shared/actions/deeplinks.tsx

```diff
@@ -0,0 +1,114 @@
+    case 'profile':
+      if (parts.length !== 3 && parts.length !== 4) {
+        return [
+          DeeplinksGen.createSetKeybaseLinkError({error}),
+          RouteTreeGen.createNavigateAppend({
+            path: [{props: {errorSource: 'app'}, selected: 'keybaseLinkError'}],
+          }),
+        ]
+      }
+      if (parts[1] === 'new-proof') {
+        return [
+          parts.length === 4 && parts[3] ? ProfileGen.createUpdateUsername({username: parts[3]}) : null,
+          ProfileGen.createAddProof({platform: parts[2], reason: 'appLink'}),
+        ]
+      }
+      break
+    // Fall-through
+    case 'private':
+    case 'public':
+    case 'team':
+      return [
+        RouteTreeGen.createSwitchTab({tab: Tabs.fsTab}),
+        RouteTreeGen.createNavigateAppend({
+          path: [{props: {path: '/keybase/' + parts.join('/')}, selected: 'main'}],
+        }),
+      ]
+    case 'chat':
+      if (parts.length !== 2) {
+        return [
+          DeeplinksGen.createSetKeybaseLinkError({error}),
+          RouteTreeGen.createNavigateAppend({
+            path: [{props: {errorSource: 'app'}, selected: 'keybaseLinkError'}],
+          }),
+        ]
+      }
+      if (parts[1].includes('#')) {
+        const teamChat = parts[1].split('#')
+        if (teamChat.length !== 2) {
+          return [
+            DeeplinksGen.createSetKeybaseLinkError({error}),
+            RouteTreeGen.createNavigateAppend({
+              path: [{props: {errorSource: 'app'}, selected: 'keybaseLinkError'}],
+            }),
+          ]
+        }
+        const [teamname, channelname] = teamChat
+        return [
+          RouteTreeGen.createSwitchTab({tab: Tabs.chatTab}),
+          ChatGen.createPreviewConversation({channelname, reason: 'appLink', teamname}),
+        ]
+      } else {
+        return [
+          RouteTreeGen.createSwitchTab({tab: Tabs.chatTab}),
+          ChatGen.createPreviewConversation({participants: parts[1].split(','), reason: 'appLink'}),
+        ]
+      }
+    default:
+      return [
+        DeeplinksGen.createSetKeybaseLinkError({error}),
+        RouteTreeGen.createNavigateAppend({
+          path: [{props: {errorSource: 'app'}, selected: 'keybaseLinkError'}],
+        }),
+      ]
+  }
+  return undefined
+}
+
+const handleAppLink = (_: Container.TypedState, action: DeeplinksGen.LinkPayload) => {
+  if (action.payload.link.startsWith('web+stellar:')) {
+    return WalletsGen.createValidateSEP7Link({link: action.payload.link})
+  } else if (action.payload.link.startsWith('keybase://')) {
+    const link = action.payload.link.replace('keybase://', '')
+    return DeeplinksGen.createHandleKeybaseLink({link})
+  } else {
+    // Normal deeplink
+    const url = new URL(action.payload.link)
+    const username = Constants.urlToUsername(url)
+    if (username) {
+      return [
+        RouteTreeGen.createNavigateAppend({path: [Tabs.peopleTab]}),
+        ProfileGen.createShowUserProfile({username}),
+      ]
+    }
```

Comment:
This is for https links on mobile right? I wonder if it'd make sense to support stuff like https://links.keybase.io/chat https://links.keybase.io/team/keybase/file here as well since we already have the code handling those in the app.

Reply:
Not sure how I feel about this.  An advantage of the protocol handler is that you don't have to tell Keybase's web server every time you're performing a Keybase interaction, and tell it what the interaction was specifically.  It seems like using the web redirect introduces some more server trust and server knowledge, and I'm not sure what we would gain in return over just using keybase://.

Reply:
Specifically, note that https://links.keybase/io/team/someteam/somefile would normally be encrypted metadata, not plaintext.  If someone shared a keybase:// link to it in a private place, it would stay private, but if someone shared a https://links.keybase.io/ link to it in a private place, it becomes disclosed to us (and e.g. a corporate TLS proxy if they are using one).

Reply:
Got it; good point! I recall briefly discussing this with Max in in 1:1. No definitely answer came out of that, but one idea we liked was to encrypt the path with a PTK, and client would just decrypt that to get the real path. We'd need to add a KID, which shortens the available space in the URL, but should still work for most paths.

Either way, for chat and profile it'd be an easier story. Perhaps we can support those as a first step? (Not suggesting we should do it in this PR; just thinking out loud about ideas.)

Reply:
@songgao I guess I feel like the only reason we would bother with all this work is because `keybase://` didn't exist before.  But now it does, so why not just use it?

Reply:
I'm not saying we should use https instead of keybase://. I agree we should use it on desktop. The reason why I brought it up was (I thought) we can't use `keybase://` on mobile, but only https://. Is that not true?

Reply:
(and I'm realizing I didn't mention mobile in my first comment. Sorry!)

Reply:
@songgao Not true!  `keybase://` works just the same on mobile after this PR :)

Reply:
nice! 🎉